osu.Game/Beatmaps/BeatmapDifficultyCache.cs

```diff
@@ -31,7 +31,7 @@ public class BeatmapDifficultyCache : MemoryCachingComponent<BeatmapDifficultyCa
         private readonly ThreadedTaskScheduler updateScheduler = new ThreadedTaskScheduler(1, nameof(BeatmapDifficultyCache));
 
         // All bindables that should be updated along with the current ruleset + mods.
-        private readonly LockedWeakList<BindableStarDifficulty> trackedBindables = new LockedWeakList<BindableStarDifficulty>();
+        private readonly WeakList<BindableStarDifficulty> trackedBindables = new WeakList<BindableStarDifficulty>();
```

Comment:
Is this change actually required? The issue looks to have been specific to `linkedCancellationSources`.

Reply:
Well it isn't, but I'm locking the entire process anyway and thought it was better to remove the double-lock.

Reply:
Fair enough, just wanted to make sure I wasn't missing something.